src/BenchmarksApps/TodosApi/Program.cs

```diff
@@ -22,20 +27,21 @@
     .AddSimpleConsole();
 #endif
 
-// Configure authentication & authorization
-builder.Services.AddAuthentication()
-    .AddJwtBearer(JwtConfiguration.ConfigureJwtBearer(builder));
+// Bind app settings from configuration & validate
+builder.Services.ConfigureAppSettings(builder.Configuration);
 
+// Configure authentication & authorization
+builder.Services.AddAuthentication().AddJwtBearer();
+builder.Services.ConfigureOptions<JwtConfiguration>();
 builder.Services.AddAuthorization();
 
 // Configure data access
-var connectionString = builder.Configuration.GetConnectionString("TodoDb")
-    ?? throw new InvalidOperationException("""
-        Connection string not found.
-        If running locally, set the connection string in user secrets for key 'ConnectionStrings:TodoDb'.
-        If running after deployment, set the connection string via the environment variable 'CONNECTIONSTRINGS__TODODB'.
-        """);
-builder.Services.AddSingleton(_ => new NpgsqlSlimDataSourceBuilder(connectionString).Build());
+builder.Services.AddSingleton(sp =>
+{
+    var appSettings = sp.GetRequiredService<IOptions<AppSettings>>().Value;
+    return new NpgsqlSlimDataSourceBuilder(appSettings.ConnectionString).Build();
+});
+builder.Services.AddHostedService<DatabaseInitializer>();
```

Comment:
Random thought:

I kind of liked how it was done before - called inline at the beginning of the app. Disadvantages of making this a "hosted service" that I see are:

1. It is only needed at startup. Holding onto the object for the lifetime of the app just to call a "no-op" Stop method seems wasteful.
2. There are ordering concerns such that this "initialization" might not have happened by the time some other piece of the app needs it. Doing it inline makes it explicit when during startup this happens.

Reply:
I moved it here in an attempt to have it not run when the build action that generates the OpenAPI document runs. It turns out the way that works is more intrusive than I imagined and it actually fully starts the application (during build!) but with a fake `IServer` and `IHostLifetime` so moving it to a hosted service wasn't enough on its own. I was trying to find the right abstraction to hook into to ensure the DB initialization happens **before** the server starts, but **not** if the app is started by an external tool doing host inspection (e.g. EF migrations, OpenAPI generation, etc.) but this might not be the right/ideal solution. It might actually not be possible given the way all the pieces interact right now.